pkg/util/kernel/headers.go

```diff
@@ -0,0 +1,157 @@
+// +build linux
+
+package kernel
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/metadata/host"
+	"github.com/mholt/archiver/v3"
+)
+
+const sysfsHeadersPath = "/sys/kernel/kheaders.tar.xz"
+const kernelModulesPath = "/lib/modules/%s/build"
+
+func FindHeaderDirs() ([]string, error) {
```

Comment:
It would be useful to explain in a comment which branches of this get triggered when -- e.g. when the kernelModules path has the headers, when the sysfsHeadersPath is populated, when modprobe gets used, and anything similar